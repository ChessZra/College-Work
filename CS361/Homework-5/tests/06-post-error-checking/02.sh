#! /bin/bash

# this test: posting more than 1024 bytes of random data

PORT=$@

SIZE=1361
RAND_DATA=`dd if=/dev/urandom bs=$SIZE count=1 status=none | base64 | head -c $SIZE`

EOL=$'\r\n'

REQUEST1=$'POST /write HTTP/1.1\r
Content-Length: '${SIZE}${EOL}${EOL}${RAND_DATA}

REQUEST2=$'GET /read HTTP/1.1'${EOL}${EOL}

printf "$REQUEST1" | nc -N 127.0.0.1 $PORT
printf "\n"
printf "$REQUEST2" | nc -N 127.0.0.1 $PORT
