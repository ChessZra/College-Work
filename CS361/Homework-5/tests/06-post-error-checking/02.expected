HTTP/1.1 413 Request Entity Too Large


HTTP/1.1 200 OK
Content-Length: 7

<empty>